### PR TITLE
Make sure ABR state stays in sync with DB state

### PR DESCRIPTION
**Pull request by stuartwdouglas**
The DB recieved code was only updating a single owner, however an ABR can have many owners. For the 'happy path' this was covered by a reconciliation with state Building which would move the state to complete.

If the AB had failed due to contamination however only a single AB would be updated to complete, which is why a lot of the Netty ABs are reported as failing even though the build has completed.

### CODE REVIEW

File: pkg/reconciler/artifactbuild/artifactbuild.go [hunk 1]
```diff
@@ -238,39 +238,43 @@ func (r *ReconcileArtifactBuild) handleDependencyBuildReceived(ctx context.Conte
...
+			key := types.NamespacedName{Namespace: db.Namespace, Name: ownerName}
+			abr := v1alpha1.ArtifactBuild{}
+			err := r.client.Get(ctx, key, &abr)
+			if err != nil {
+				msg := "get for dependencybuild %s:%s owning abr %s:%s yielded error %s"
+				r.eventRecorder.Eventf(db, corev1.EventTypeWarning, msg, db.Namespace, db.Name, db.Namespace, ownerName, err.Error())
+				log.Error(err, fmt.Sprintf(msg, db.Namespace, db.Name, db.Namespace, ownerName, err.Error()))
+				return reconcile.Result{}, err
+			}
+			switch db.Status.State {
```
[the remark this answers is not in the document]
Comment (stuartwdouglas) [reply pkg/reconciler/artifactbuild/artifactbuild.go:262]: Done

**Comment by stuartwdouglas** [comment]
Looking at the resulting report all Netty ABs now end up in the correct state.

**Comment by gabemontero** [comment]
I rebased this on the just merged periodic fix of yours @stuartwdouglas 

/test jvm-build-service-in-repo-apicurio-e2e

but if that flakes in unrelated ways, I'll merge anyway based on your prior findings at https://github.com/redhat-appstudio/jvm-build-service/pull/298#issuecomment-1269366838 and https://github.com/redhat-appstudio/jvm-build-service/pull/298#issuecomment-1269368044

**Pull request merged**